docs/data/material/guides/minimizing-bundle-size/minimizing-bundle-size.md

```diff
@@ -214,47 +214,7 @@ It will perform the following diffs:
 
 The packages published on npm are **transpiled** with [Babel](https://github.com/babel/babel), optimized for performance with the [supported platforms](/material-ui/getting-started/supported-platforms/).
 
-A [modern bundle](#modern-bundle) is also available.
-
-### How to use custom bundles?
-
-:::error
-You are strongly discouraged to:
-
-- Import from any of the custom bundles directly. Do not do this:
-
-  ```js
-  import { Button } from '@mui/material/legacy';
```

Comment:
I think this section is still valuable. We need to keep documenting that these are not public API. I'd just replace `/legacy` with `/modern` here.

Somewhere in this page we also recommend `babel-plugin-direct-import`. It's harmful as that plugin does not respect what we export as public API. intead I think we should remove that recommendation and instead add it to this list with patterns to avoid.

We could also consider mentioning Next.js [`optimizePackageImports`](https://nextjs.org/docs/app/api-reference/next-config-js/optimizePackageImports) as an alternative to those babel plugins.

Reply:
> I think this section is still valuable. We need to keep documenting that these are not public API. I'd just replace `/legacy` with `/modern` here.

Got it, I'll make that change.

> Somewhere in this page we also recommend `babel-plugin-direct-import`. It's harmful as that plugin does not respect what we export as public API. intead I think we should remove that recommendation and instead add it to this list with patterns to avoid.

You mean like the example in their [documentation](https://github.com/avocadowastaken/babel-plugin-direct-import#example) where the exported structure (OUT example) isn't aligned with the public API? If so, why not just remove the plugin recommendation entirely?

> We could also consider mentioning Next.js [`optimizePackageImports`](https://nextjs.org/docs/app/api-reference/next-config-js/optimizePackageImports) as an alternative to those babel plugins.

From what I see in their docs, it looks like `@mui/material` is already optimized by default without this option. So, we might not need to mention it separately.

Reply:
> why not just remove the plugin recommendation entirely?

Yes, but I meant we could also consider advising against it. Maybe 🤔 

> it looks like @mui/material is already optimized by default 

👍

Reply:
@Janpot I replaced `legacy` with `modern`. As for `babel-plugin-direct-import`, I’m not sure we should advise against it. Based on [its tests](https://github.com/avocadowastaken/babel-plugin-direct-import/blob/main/test/%40mui/material/exports.spec.js), it seems to respect our public API.

Reply:
For some reason they're including the `.js` extension, we don't support this and it will break in a next major. We could also open an issue with them first and remove our recommendation if they don't change it?

Reply:
> For some reason they're including the .js extension, we don't support this

I'm sorry, I’m not sure I understand. From what I can see, we have `.js` files in the `build` folder. Could you explain for my understanding?

Reply:
We may have them in the build folder, but that doesn't mean we consider using the extension as public API. Especially in a future version where we will [package.json `exports`](https://nodejs.org/api/packages.html#conditional-exports), importing these will start to fail, even though the files exist. Having these extensions also prevents bundlers and runtimes from resolving to the correct format (esm or commonjs) depending on the request.

Reply:
Got it, thanks for the explanation.